yalafi/packages/cleveref.py

```diff
@@ -0,0 +1,215 @@
+
+#
+#   YaLafi module for LaTeX package cleveref
+#
+
+from yalafi.defs import InitModule, Macro, VoidToken
+from yalafi import utils
+import re
+
+
+require_packages = []
+
+# Regular Expressions used to read sed file:
+re_ref = re.compile(r'''
+s/\\
+(                   # group 1: '\cref' or '\Cref'
+    \\cref|\\Cref
+)
+(?:\\)?             # non capturing group
+(\*?)               # group 2: '*' if present, '' else
+\{
+    ([^\}\{]+)      # group 3: the label of the reference
+\}
+/
+    (.*)            # group 4: the replacement string
+/g
+''', re.VERBOSE)
+re_ref_range = re.compile(r'''
+s/\\
+(                   # group 1: '\cref' or '\Cref'
+    \\crefrange|\\Crefrange
+)
+(?:\\)?             # non capturing group
+(\*?)               # group 2: '*' if present, '' else
+\{
+    ([^\}\{]+)      # group 3: the label of the reference
+\}
+/

```

Comment:
Hopefully, I'm wrong. But are you missing to match the second argument enclosed in {}?

Reply:
Yes. Thanks for spotting this one. I forgot to add it when I changed to the `re.VERBOSE` style. I am just wondering why the test passed.

Reply:
The reason is probably that you did not raise an error message in `h_make_crefrange()`.

Reply:
You are right. I just checked: With the updated function `f` in `h_make_crefrange()` the test fails because `LATEXXXERROR` is added to the text.